pkg/pod/status.go

```diff
@@ -116,6 +118,15 @@ func MakeTaskRunStatus(logger *zap.SugaredLogger, tr v1alpha1.TaskRun, pod *core
 	trs.Steps = []v1alpha1.StepState{}
 	trs.Sidecars = []v1alpha1.SidecarState{}
 
+	// Only set this once, the first time.
+	if trs.TaskSpec == nil {
+		ts := v1beta1.TaskSpec{}
+		if err := taskSpec.ConvertTo(context.Background(), &ts); err != nil {
+			logger.Errorf("error setting taskrun.Status.taskSpec in taskrun %s: %w", tr.Name, err)
+		}
+		trs.TaskSpec = &ts
+	}
+
```

Comment:
I till think that https://github.com/tektoncd/pipeline/blob/9344e4f2291bb5b9a35f50f7549c520e02c8bf56/pkg/reconciler/taskrun/taskrun.go#L207-L208 would be a better place for this.
Some tasks will never have a pod successfully created, and for them there will be no taskspec available for audit.

Still this works fine for many other cases, so I don't mind go ahead with this and adding on top if you prefer so.

Something that I would like to do is to use the spec stored in there during reconcile as well, but since that would break `cancel`, we need to find a solution to that first.

Reply:
Ah, that makes a lot of sense. Thanks for explaining! I made the change.